framework/src/modules/chain/logic/bft/bft.js

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const HeadersList = require('./headers_list');
+const blockHeaderSchema = require('./block_header_schema');
+const { validate } = require('../../../../../src/controller/validator');
+
+/**
+ * @typedef {Object} BlockHeader
+ * @property {string} blockId
+ * @property {int} height
+ * @property {int} maxHeightPreviouslyForged
+ * @property {int} prevotedConfirmedUptoHeight
+ * @property {int} activeSinceRound
+ * @property {string} delegatePublicKey
+ */
+
+class BFT {
+	constructor({ finalizedHeight, activeDelegates = 101 }) {
+		// Set constants
+		this.ACTIVE_DELEGATES = activeDelegates;
```

Comment:
Prefer not to use capital snake, since this is not real constant

Reply:
I am not sure about introducing 101 as default here. I think it would be better if we set it to default in the configuration layer to 101 and we pass the value from there to the constructor.

Reply:
I am not sure about introducing 101 as default here. I think it might be better if we set it to default in the configuration layer to 101 and we pass the value from there to the constructor.

Reply:
@limiaspasdaniel Resolved.

Reply:
@shuse2 I used convention for readonly attributes. Suggest if you don't like this.

Reply:
since this is normal config value, i would prefer to keep it like `this.activeDelegates` or even make it more meaningful like `numOfActiveDelegates`

Reply:
This is not addressed yet, but other than this it looks good

Reply:
Done on your Wishlist.